packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	__experimentalGetBlockLabel as getBlockLabel,
+	getBlockType,
+} from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+
+function useSecondaryText() {
+	const selectedBlock = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSelectedBlock();
+	} );
+
+	// TODO: Handle if parent is template part too.
+	const selectedBlockLabel =
+		selectedBlock?.name === 'core/template-part'
+			? getBlockLabel(
+					getBlockType( selectedBlock?.name ),
+					selectedBlock?.attributes
+			  )
+			: null;
+
+	if ( selectedBlockLabel ) {
+		return {
+			label: selectedBlockLabel,
+			isActive: true,
+		};
+	}
+	return {};
+}
+
+export default function DocumentActions( { documentTitle } ) {
+	const { label, isActive } = useSecondaryText();
```

Comment:
why do we need `isActive` ?   `isActive` seems to always be true when there is a label, and undefined when the label is undefined.  

Do we need a fallback/placeholder name for when `selectedBlockLabel` is an empty string or something?

Reply:
I was thinking ahead to hover state, where if you were hovering the item, it might be grayed out until you select it. But that's TBD I guess, so it doesn't do a whole lot right now!